pkg/database/realm.go

```diff
@@ -1159,3 +1160,39 @@ func ToCIDRList(s string) ([]string, error) {
 	sort.Strings(cidrs)
 	return cidrs, nil
 }
+
+// RealmUserStats carries the per-user-per-day-per-realm Codes issued.
+// This is a structure joined from multiple tables in the DB.
+type RealmUserStats struct {
+	UserID      uint
+	Name        string
+	CodesIssued uint
+	Date        time.Time
+}
+
+// CodesPerUser returns a set of UserStats for a given date range.
+func (r *Realm) CodesPerUser(db *Database, start, stop time.Time) ([]*RealmUserStats, error) {
+	start = start.Truncate(time.Hour * 24)
+	stop = stop.Truncate(time.Hour * 24)
+	if start.After(stop) {
+		return nil, ErrBadDateRange
+	}
+
+	var stats []*RealmUserStats
+	if err := db.db.
+		Model(&UserStats{}).
+		Select("users.id, users.name, codes_issued, date").
+		Where("realm_id = ?", r.ID).
+		Where("date >= ? AND date <= ?", start, stop).
+		Joins("INNNER JOIN users ON users.id = user_id").
+		Order("date ASC").
```

Comment:
Did you verify this actually works? I've had weird issues with Gorm where the order call has to come before select/where.

Reply:
Added to test case. PTAL